contentful/apps
# feat: authorize ms teams on config page [INTEG-1474]

## Purpose

This PR creates a way for users to sign in to Microsoft so that we can use the Microsoft Graph API.

## Approach

Uses the [Microsoft Authentication Library for React](https://www.npmjs.com/package/@azure/msal-react) to handle authentication. Users can click a button to sign in, they login with their credentials through the popup. Once successfully logged in, they see their MS name and a Disconnect button.

<img width="926" alt="Screenshot 2023-10-30 at 8 13 24 AM" src="https://github.com/contentful/apps/assets/62958907/d116878e-d7df-4994-b842-3e85dfd42a56">

<img width="924" alt="Screenshot 2023-10-30 at 8 13 14 AM" src="https://github.com/contentful/apps/assets/62958907/12ffe5ba-ae08-440f-9bfc-491108e5fe7c">

## Testing steps

Run the app locally and add the correct credentials in `.env.development`. You should be able to utilize this flow to log into both our sandbox account and a work account.

## Breaking Changes

## Dependencies and/or References

Helpful guide for [MSAL](https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-single-page-app-react-sign-in).

## Deployment
